### PR TITLE
[HAL-1511] - Cross-site scripting (XSS) in JBoss Management Console

### PR DESCRIPTION
**Pull request by adamecp**
Jira task [https://issues.jboss.org/browse/HAL-1511](https://issues.jboss.org/browse/HAL-1511)
Description: There is a vulnerability in HAL not properly escaping html characters from resource names

PR to branch master [https://github.com/hal/testsuite/pull/379](https://github.com/hal/testsuite/pull/379)

### CODE REVIEW

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Before
+    public void before() {
+        navigation = new FinderNavigation(browser, StandaloneDeploymentEntryPoint.class);
+    }
+
+
+    @BeforeClass
+    public static void setUp() throws IOException, TimeoutException, InterruptedException, CommandFailedException {
+        administration.reloadIfRequired();
```
Comment (spriadka) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java]: Is the `administration.reloadIfRequired()` necessary before deployment?. If not I suggest calling this method after the deployment.

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                .name(NAME)
+                .build());
+    }
+
+    @AfterClass
+    public static void cleanUp() throws IOException, CommandFailedException, OperationException, TimeoutException, InterruptedException {
+        try {
+            ops.undeployIfExists(SHORT_NAME);
+            administration.restartIfRequired();
+            administration.reloadIfRequired();
```
Comment (spriadka) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java:61]: Are both `reload` and `restart` needed after the undeploy? If not, I suggest to `reload` only.

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    public void before() {
+        navigation = new FinderNavigation(browser, StandaloneDeploymentEntryPoint.class);
+    }
+
+
+    @BeforeClass
+    public static void setUp() throws IOException, TimeoutException, InterruptedException, CommandFailedException {
+        administration.reloadIfRequired();
+
+        client.apply(new DeployCommand.Builder(FILE_PATH + FILE_NAME)
```
Comment (spriadka) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java]: I suggest replacing the `FILE_PATH + FILE_NAME` with `getClass().getResource()` syntax, as files present in the `src/test/resources` are available through this methods

File: basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    WebDriver browser;
+
+    @Before
+    public void before() {
+        navigation = new FinderNavigation(browser, StandaloneDeploymentEntryPoint.class);
+    }
+
+
+    @BeforeClass
+    public static void setUp() throws IOException, TimeoutException, InterruptedException, CommandFailedException {
```
Comment (spriadka) [inline basic/src/test/java/org/jboss/hal/testsuite/test/runtime/deployments/XssScriptingTestCase.java:51]: I suggest moving static methods & fields before instance methods & fields

**Review by spriadka — CHANGES_REQUESTED** [review]
Suggested some minor changes before merging, otherwise LGTM, thanks for the contribution @adamecp

**Comment by spriadka** [comment]
Once this PR is resolved, feel free to merge #379 as well.

**Review by adamecp — COMMENTED** [review]
I rewrote it

**Comment by spriadka** [comment]
Thanks @adamecp :+1: , just squash a bit and I'll merge it.

**Review by spriadka — APPROVED** [review]

**Pull request merged**